repo/blob/throttling/token_bucket.go

```diff
@@ -0,0 +1,124 @@
+package throttling
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/repo/logging"
+)
+
+var log = logging.Module("throttling")
+
+type tokenBucket struct {
+	name  string
+	now   func() time.Time
+	sleep func(ctx context.Context, d time.Duration)
+
+	mu                   sync.Mutex
+	lastTime             time.Time
+	numTokens            float64
+	maxTokens            float64
+	addTokensPerTimeUnit float64
```

Comment:
is there any case where `addTokensPerTimeUnit != maxTokens` ?

Reply:
not really, removed for now